src/binary-reader.cc

```diff
@@ -1605,6 +1605,9 @@ static void read_custom_section(Context* ctx, uint32_t section_size) {
       uint32_t subsection_size;
       in_u32_leb128(ctx, &name_type, "name type");
       in_u32_leb128(ctx, &subsection_size, "subsection size");
+      size_t subsection_end = ctx->offset + subsection_size;
+      if (subsection_end > ctx->read_end)
+        RAISE_ERROR("invalid sub-section size: extends past end");
```

Comment:
Actually I just tested the current "master" and it already blows up on the old style name section:

```
$ ./out/wasmdump  -x ./debug-names-1.wasm
*ERROR*: @0x0000003d: unable to read string: name
```

So this change doesn't actually make it worse.